ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -393,19 +403,25 @@ func (r *RayJobReconciler) getSubmitterTemplate(rayJobInstance *rayv1.RayJob) (v
 }
 
 // createNewK8sJob creates a new Kubernetes Job. It returns the Job's name and a boolean indicating whether a new Job was created.
-func (r *RayJobReconciler) createNewK8sJob(ctx context.Context, rayJobInstance *rayv1.RayJob, submitterTemplate v1.PodTemplateSpec, rayClusterInstance *rayv1.RayCluster) (string, bool, error) {
+func (r *RayJobReconciler) createNewK8sJob(ctx context.Context, rayJobInstance *rayv1.RayJob, submitterTemplate v1.PodTemplateSpec) (string, bool, error) {
 	job := &batchv1.Job{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      rayJobInstance.Name,
 			Namespace: rayJobInstance.Namespace,
+			Labels: map[string]string{
+				common.KubernetesCreatedByLabelKey: common.ComponentName,
+			},
 		},
 		Spec: batchv1.JobSpec{
-			Template: submitterTemplate,
+			// The `ray job submit` command fails when submission is retried with the same job ID.
+			// Let's disable backoff retry, until job submission retry effectively works.
+			BackoffLimit: pointer.Int32(0),
+			Template:     submitterTemplate,
 		},
 	}
 
 	// Set the ownership in order to do the garbage collection by k8s.
-	if err := ctrl.SetControllerReference(rayClusterInstance, job, r.Scheme); err != nil {
+	if err := ctrl.SetControllerReference(rayJobInstance, job, r.Scheme); err != nil {
```

Comment:
We might implement retry logic for RayJob in the future. If RayJob retries, it will spawn a new RayCluster and a new K8s Job. Therefore, the controller for the K8s Job should be the RayCluster.

Reply:
I think this will require more thought, depending on the evolution of the retry mechanism, and the RayCluster reuse case. Yet at the moment, as a user, I find it surprising that the Job is not cascade-deleted when I delete the RayJob.

Reply:
> Yet at the moment, as a user, I find it surprising that the Job is not cascade-deleted when I delete the RayJob.

Really? It should be deleted. If not, it is a bug. RayJob is the controller reference of the RayCluster, and the RayCluster is the controller reference of the Kubernetes Job. Hence, I expected that the K8s Job should be removed if RayJob is deleted.

Reply:
Not in the case the RayJob does not own the RayCluster, that is when it's created with a `clusterSelector` field.

Reply:
<img width="851" alt="Screen Shot 2023-10-19 at 9 53 36 AM" src="https://github.com/ray-project/kuberay/assets/20109646/bb981da3-8392-4dfe-95b3-0be376fa01e6">

I conducted an experiment, and I do not observe the behavior.

Reply:
> Not in the case the RayJob does not own the RayCluster, that is when it's created with a clusterSelector field.

Got it. We do not encourage users to use that.

Reply:
You can ignore the comment at https://github.com/ray-project/kuberay/pull/1539/files#r1365854531. I hadn't seen your reply when I wrote mine.